File: src/main/java/no/nav/veilarboppfolging/client/veilarbarena/YtelserDTO.java
package no.nav.veilarboppfolging.client.veilarbarena;

import lombok.Data;
import lombok.Value;
import lombok.experimental.Accessors;

import java.time.LocalDate;
import java.util.List;

@Value
public class YtelserDTO {
    public List<Vedtak> vedtak;
    public List<Ytelseskontrakt> ytelser;

    @Data
    @Accessors(chain = true)
    public static class Vedtak {
        public String type;
        public String status;
        public String aktivitetsfase;
        public String rettighetsgruppe;
        public LocalDate fraDato;
        public LocalDate tilDato;
    }

    @Data
    @Accessors(chain = true)
    public static class Ytelseskontrakt {
        public String type;
        public String status;
        public LocalDate motattDato;
        public LocalDate fraDato;
        public LocalDate tilDato;
    }
}

File: src/main/kotlin/no/nav/veilarboppfolging/client/veilarbarena/YtelserDTO.kt
package no.nav.veilarboppfolging.client.veilarbarena

import java.time.LocalDate


data class YtelserDTO(
    val vedtak: List<Vedtak>,
    var ytelser: List<Ytelseskontrakt>,
)

data class Vedtak (
    var type: String?,
    var status: String?,
    var aktivitetsfase: String?,
    var rettighetsgruppe: String?,
    var fraDato: LocalDate?,
    var tilDato: LocalDate?,
)

data class Ytelseskontrakt (
    var type: String?,
    var status: String?,
    var motattDato: LocalDate?,
    var fraDato: LocalDate?,
    var tilDato: LocalDate?,
)

